packages/styled-components/src/models/StyledNativeComponent.js

```diff
@@ -32,43 +125,42 @@ class StyledNativeComponent extends Component<*, *> {
             $as: transientAsProp,
             as: renderAs,
             forwardedComponent,
-            forwardedAs,
-            forwardedRef,
-            style = [],
             ...props
           } = this.props;
 
-          const { defaultProps, target, shouldForwardProp } = forwardedComponent;
+          const { defaultProps, target } = forwardedComponent;
           const elementToBeRendered =
             this.attrs.$as || this.attrs.as || transientAsProp || renderAs || target;
 
-          const generatedStyles = this.generateAndInjectStyles(
+          const generatedStyles = {...this.generateAndInjectStyles(
```

Comment:
Is the `{...` required for it to work? In my attempt to get this rendering with React Primitives, this appears to be breaking the render.

Reply:
This is required because we will edit the object and overwrite the values. Check below for instance:
`this.remStyles.forEach(key => generatedStyles[key]*=16)`

I think that removing this could generate problematic border effects...

Reply:
Can you tell me what the issue is? I don't see why it would break the render... It is the same object in the end... Unless there are some shady operations involved?

Reply:
Ah, ok. If it's working fine in React Native, should be fine to leave it as is. I was trying to get it working with `react-sketchapp` (primitives platform) and it was breaking here, but that use-case is probably outside the scope of this PR anyway.